analysis/src/Hint.ml

```diff
@@ -119,4 +119,58 @@ let inlay ~path ~pos ~maxLength ~debug =
                | Some value ->
                  if String.length label > value then None else Some result
                | None -> Some result)
-             | None -> None)))
\ No newline at end of file
+             | None -> None)))
+
+let codeLens ~path ~debug =
+  let lenses = ref [] in
+  let push loc =
+    let range = Utils.cmtLocToRange loc in
+    lenses := range :: !lenses
+  in
+  (* Code lenses are only emitted for functions right now. So look for value bindings that are functions,
```

Comment:
What about interface files?

Reply:
Not emitted for interface files, they're redundant there imo since the definition in the interface file will be almost identical to what would've been printed in the code lens anyway.

Reply:
I'll add a comment on why.